packages/application-shell/src/test-utils/custom-views-utils.tsx

```diff
@@ -0,0 +1,107 @@
+import { type ReactNode, useEffect, useRef } from 'react';
```

Comment:
This new file contains helpers specific for Custom Views tests so we don't have to include them in the template so we keep control over them.